src/models/event-timeline-set.js

```diff
@@ -166,21 +170,37 @@ EventTimelineSet.prototype.resetLiveTimeline = function(backPaginationToken, flu
         newTimeline = this.addTimeline();
     }
 
-    // initialise the state in the new timeline from our last known state
-    const evMap = this._liveTimeline.getState(EventTimeline.FORWARDS).events;
-    const events = [];
-    for (const evtype in evMap) {
-        if (!evMap.hasOwnProperty(evtype)) {
-            continue;
-        }
-        for (const stateKey in evMap[evtype]) {
-            if (!evMap[evtype].hasOwnProperty(stateKey)) {
+    // move the state objects over from the old live timeline, then we'll
+    // keep using the same RoomMember objects for the 'live' set of members
+    newTimeline._startState = this._liveTimeline._startState;
```

Comment:
shouldn't this be `this._liveTimeline._endState`? Or rather, a new copy thereof?